contracts/base/LockAndMsgSender.sol

```diff
@@ -1,35 +1,27 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 pragma solidity ^0.8.24;
 
-import {Constants} from '../libraries/Constants.sol';
+import {Locker} from '../libraries/Locker.sol';
 
 contract LockAndMsgSender {
     error ContractLocked();
 
-    address internal constant NOT_LOCKED_FLAG = address(1);
-    address internal lockedBy = NOT_LOCKED_FLAG;
-
+    /// @notice Modifier enforcing a reentrancy lock that allows self-reentrancy
     modifier isNotLocked() {
+        // The contract is allowed to reenter itself to perform EXECUTE_SUB_PLAN commands
```

Comment:
tbh I'm not really sure what this means/what is the motivation for this comment

Reply:
Just trying to explain the modifier because it performs no checks in the case where `msg.sender == address(this)`. I'll rewrite it a bit